backend/fbdev.c

```diff
@@ -22,6 +22,25 @@
 #define SCREEN(x) ((twin_context_t *) x)->screen
 #define PRIV(x) ((twin_fbdev_t *) ((twin_context_t *) x)->priv)
 
+/* color conversion */
+#define ARGB32_TO_RGB565_PERLINE(dest, pixels, width)   \
+    do {                                                \
+        for (int i = 0; i < width; i++)                 \
+            dest[i] = ((pixels[i] & 0x00f80000) >> 8) | \
+                      ((pixels[i] & 0x0000fc00) >> 5) | \
+                      ((pixels[i] & 0x000000f8) >> 3);  \
+    } while (0)
+
+/* Requires validation in 24-bit per pixel environments */
+#define ARGB32_TO_RGB888_PERLINE(dest, pixels, width) \
+    do {                                              \
+        for (int i = 0; i < width; i++)               \
+            dest[i] = 0xff000000 | pixels[i];         \
+    } while (0)
```

Comment:
@Bennctu, can you comment on the above macros for utilizing Pixman for faster conversions?

Reply:
[Pixman API](https://gitlab.freedesktop.org/pixman/pixman/-/blob/master/pixman/pixman.h?ref_type=heads) does not support color conversion.